code/script.js

```diff
@@ -225,101 +233,197 @@ const generateBoard = () => {
+      alert(
+        `Yes, the person has ${value} eyes! Keep all people with ${value} eyes`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert(
+        `No, the person doesn't have ${value} eyes! Remove all people with ${value} eyes`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
     }
   }
+  guesses++
+  guessCounter.innerHTML = guesses
+  generateBoard()
+  } 
+  
+  
 
   // Determine what is the category
   // filter by category to keep or remove based on the keep variable.
   /* 
-    for hair and eyes :
+    for Hair and Eyes :
       charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
       or
       charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
 
-    for accessories and other
+    for Accessories and Other
       charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
       or
       charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
   */
 
   // Invoke a function to redraw the board with the remaining people.
-}
+
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
+  const player = confirm(`Are you sure you want to guess?`)
+   if (player) {
+     checkMyGuess(personToConfirm)
+   }
   // store the interaction from the player in a variable.
   // remember the confirm() ?
   // If the player wants to guess, invoke the checkMyGuess function.
 }
 
 // If you confirm, this function is invoked
 const checkMyGuess = (personToCheck) => {
+  if (personToCheck === secret.name) {
+    console.log(`toimii`)
+    winOrLoseText.innerHTML = `<h1>Your guess is correct!</h1>`
+
+  } else {
+    console.log(`toimii2`)
+    winOrLoseText.innerHTML = `<h1>Sorry, wrong answer!</h1>`
+  }
+  board.innerHTML = ''
+  winOrLose.style.display='block'
+  gameTimer.style.display= 'none'
+  gameTime.style.display = 'none'
+  guessCounter.style.display = 'none'
+  gameGuess.style.display = 'none'
+}
+
+
   // 1. Check if the personToCheck is the same as the secret person's name
   // 2. Set a Message to show in the win or lose section accordingly
   // 3. Show the win or lose section
   // 4. Hide the game board
-}
+
 
 // Invokes the start function when website is loaded
 start()
 
 // All the event listeners
-restartButton.addEventListener('click', start)
+restartButton.addEventListener('click', () => {
+  start()
+})
+
+questions.addEventListener('change', () => {
+   selectQuestion()
+})
+
+findOutButton.addEventListener('click', () => {
+  checkQuestion()
+})
+
+playAgainButton.addEventListener('click', () => {
+  start()
```

Comment:
You can just write it like `restartButton.addEventListener('click', start)` insted of 
```
restartButton.addEventListener('click', () => {
 start()
```
you don't need to add the arrow and curly brackets :)